main.go

```diff
@@ -41,34 +43,50 @@ var (
 func main() {
 	ctx := context.Background()
 
-	start := time.Now()
-
 	// Overloads `cli` with configuration file values.
 	cmd := kong.Parse(&cli,
 		kong.Vars{"version": version, "default_config_path": defaultConfigPath},
 		kong.NamedMapper("yamlfile", kongyaml.YAMLFileMapper),
 		kong.Configuration(kongyaml.Loader),
 		kong.BindTo(ctx, (*context.Context)(nil)))
 
+	err := Run(ctx, cmd)
+	cmd.FatalIfErrorf(err)
+}
+
+func Run(ctx context.Context, cmd *kong.Context) error {
+	start := time.Now()
+
 	// check the token is set
 	if cli.Token == "" {
 		log.Fatal().Msg("missing token, please set the BUILDKITE_AGENT_ACCESS_TOKEN environment variable")
 	}
 
 	tp, err := trace.NewProvider(ctx, cli.TraceExporter, "github.com/buildkite/zstash", version)
 	if err != nil {
-		log.Fatal().Err(err).Msg("failed to create trace provider")
+		return fmt.Errorf("failed to create trace provider: %w", err)
 	}
 	defer func() {
 		_ = tp.Shutdown(ctx)
 	}()
 
-	ctx, span := trace.Start(ctx, "zstash")
-	defer span.End()
-
```

Comment:
Moving this into a command and separating the core functionality out, am I right that the thought here is the library content will all provide the spans as needed and the command therefore won't have the additional layer?